pulsar-client/src/main/java/org/apache/pulsar/client/api/Message.java

```diff
@@ -62,6 +63,8 @@
      */
     byte[] getData();
```

Comment:
Here we could update Javadoc saying that this would return the serialized form of the data.

One thing to consider, probably later though before 2.0, is that when we do the deserialization, we could already release the netty buffer (that would keep all the bytes in pooled direct memory). The only issue is at that point, the `getData()` won't be available.

Would it make sense to either do : 
 * `T getData()`
  * or deprecate `getData()` and just leave `getEvent()` / `getValue()` or similar.